astrocut/make_cube.py

```diff
@@ -346,3 +349,500 @@ def make_cube(self, file_list, cube_file="img-cube.fits", sector=None, max_memor
+
+        # Working out the block size and number of blocks needed for writing the cube
+        # without using too much memory
+        slice_size = image_shape[1] * len(self.file_list) * 2 * 4  # in bytes (float32)
+        max_block_size = int((self.max_memory * 1e9)//slice_size)
+        
+        self.num_blocks = int(image_shape[0]/max_block_size + 1)
+        self.block_size = int(image_shape[0]/self.num_blocks + 1)
+        
+        # Determining cube shape: 
+        # If it's a new cube, the shape is (nRows, nCols, nImages, 2)
+        if not self.update:
+            self.cube_shape = (image_shape[0], image_shape[1], len(self.file_list), 2)
+            
+        # Else, if it's an update to an existing cube, the shape is (nRows, nCols, nImages + nNewImages, 2)
+        else: 
+            self.cube_shape = self.cube_append.shape
+            
+        # Making the primary header if there's no cube_file yet
+        if not self.update:
+            with fits.open(self.file_list[0], mode='denywrite', memmap=True) as first_file:
+                header = deepcopy(first_file[0].header)
+                header.remove('CHECKSUM', ignore_missing=True)
+
+                # Adding standard keywords
+                header['ORIGIN'] = 'STScI/MAST'
+                header['DATE'] = str(date.today())
+
+                # Adding factory specific keywords
+                for kwd in self.image_header_keywords:
+                    header[kwd] = (first_file[0].header[kwd], first_file[0].header.comments[kwd])
+
+                # Adding the extra keywords passed in
+                for kwd, value in extra_keywords.items():
+                    header[kwd] = (value[0], value[1])
+
+        # Otherwise we're updating an existing cube file
+        else:
+            with fits.open(self.cube_file, mode='update', memmap=True) as cube_hdu:
+                header = cube_hdu[0].header 
+                header['HISTORY'] = f'Updated on {str(date.today())} with new FFI delivery.' 
+                header['HISTORY'] = f'First FFI is {str(os.path.basename(self.file_list[0]))}'
+
+        # Adding the keywords from the last file
+        with fits.open(self.file_list[-1], mode='denywrite', memmap=True) as last_file:
+            for kwd in self.last_file_keywords:
+                header[kwd] = (last_file[0].header[kwd], last_file[0].header.comments[kwd])
+
+        header["EXTNAME"] = "PRIMARY"
+
+        self.primary_header = header
+
+    def _build_info_table(self):
+        """ Reading the keywords and setting up the table to hold the image headers (extension 1)
+        from every input file.
+        """
+
+        with fits.open(self.template_file, mode='denywrite', memmap=True) as ffi_data:
+            
+            # The image specific header information will be saved in a table in the second extension
+            secondary_header = ffi_data[0].header
+
+            # set up the image info table
+            already_there = []
```

Comment:
can we rename this variable `column_names` or something? and then instead of a `continue` we can just simply have a conditional before the `cols.append()`